demos/http/http_demo_s3_download/sigv4_config.h

```diff
@@ -29,45 +29,69 @@
 #ifndef SIGV4_CONFIG_H_
 #define SIGV4_CONFIG_H_
 
-#ifndef SIGV4_PROCESSING_BUFFER_LENGTH
-    #define SIGV4_PROCESSING_BUFFER_LENGTH    1024U
-#endif
+/**************************************************/
+/******* DO NOT CHANGE the following order ********/
+/**************************************************/
 
-#ifndef SIGV4_MAX_HTTP_HEADER_COUNT
-    #define SIGV4_MAX_HTTP_HEADER_COUNT    100U
-#endif
+/* Include logging header files and define logging macros in the following order:
+ * 1. Include the header file "logging_levels.h".
+ * 2. Define the LIBRARY_LOG_NAME and LIBRARY_LOG_LEVEL macros depending on
+ * the logging configuration for SIGV4.
+ * 3. Include the header file "logging_stack.h", if logging is enabled for SIGV4.
+ */
 
-#ifndef SIGV4_MAX_QUERY_PAIR_COUNT
-    #define SIGV4_MAX_QUERY_PAIR_COUNT    100U
-#endif
+#include "logging_levels.h"
 
-#ifndef SIGV4_HASH_MAX_BLOCK_LENGTH
-    #define SIGV4_HASH_MAX_BLOCK_LENGTH    64U
+/* Logging configuration for the SigV4 library. */
+#ifndef LIBRARY_LOG_NAME
+    #define LIBRARY_LOG_NAME    "SIGV4"
 #endif
 
-#ifndef SIGV4_HASH_MAX_DIGEST_LENGTH
-    #define SIGV4_HASH_MAX_DIGEST_LENGTH    32U
+#ifndef LIBRARY_LOG_LEVEL
+    #define LIBRARY_LOG_LEVEL    LOG_INFO
 #endif
 
-#ifndef SIGV4_USE_CANONICAL_SUPPORT
-    #define SIGV4_USE_CANONICAL_SUPPORT    1
-#endif
+#include "logging_stack.h"
 
-/* Logging configuration for the SigV4 library. */
-#ifndef LogError
-    #define LogError( message )
-#endif
+/**
+ * @brief Define the size of the buffer used for generating the canonical
+ * request.
+ */
+#define SIGV4_PROCESSING_BUFFER_LENGTH    2048U
```

Comment:
Why is 2048 required for the size of the buffer. Can a smaller value of 1500 not suffice?

Reply:
1500 is not sufficient. Although 1600 works updating it to 1600.